chapters/en/slides/module4/module4_01.Rmd

```diff
@@ -0,0 +1,345 @@
+the two categorical are mapped to one axis each
+and we don't need to facet.
+Sorting on color/count puts the genres with many observations close together,
+similar to how we sorted on `'x'` and `'y'` in previous modules.
+
+If we want to compare which genre is most common for a certain rating,
+we compare the colors in the same *column*.
+If we instead are interested in the most common rating assigned to a movie
+we compare the columns *row-wise*.
+We can quickly see that most dramas are rated PG-13 or R
+and most horror movies are rated R.
+
+This visualization is effective
+for quickly communicating the main takeaways from the two questions
+and giving us an overview of the data,
+but it is harder  to tell that exact count for each color
+so if that is of great importance a bar chart is more suitable.
+
+As you can see the default colorscale is different from the marks we have used so far.
+We will talk more about colorscale in a later module,
+but briefly the change in hue from one color to another
+allows for more resolution in being able to tell
+which color corresponds to a certain value.
+
+For the same reasons a density plot was not an appropriate replacement of the categorical barchart,
+we cannot replace this categorical heatmap with a 2D denity plot.
+
+---
+
+## Using both the color and marker size to indicate the count creates a more effective visualization.
+
+```{python}
+# Sort since according to the natural order of the ratings
+mpaa_ordered = ['G', 'PG', 'PG-13', 'R']
+alt.Chart(movies_extended).mark_circle().encode(
+    alt.Color('count()'),
+    alt.Size('count()'),
+    alt.X('MPAA Rating', sort=mpaa_ordered),
+    alt.Y('Major Genre', sort='x'))
+```
+
+Notes:
+TODO an alternative strategy is to not teach mark_rect for categorical and go straight to this plot... I like mentioning it to contrast it with this, but not sure if there would be confusion or if the heatmap for some reason implies continuous data.
+
+One potential concern with heatmaps
+is that they rely solely on color to communicate the value of interest.
+
+We cannot perceive small variations in color
+as accurately as we can for other visual channels,
+such as positions of size.
+
+Color can also be problematic for people with color vision deficiencies,
+which is almost 10% of the population.
+
+To ameliorate these issues,
+we can use `map_square` or `map_circle` in Altair,
+which allows us to change the size of each mark in addition to its color.
+
+This visualization is highly effective in answering both of the questions we posed initially,
+and if we wanted to,
+we could now facet by a third categorical column 
+such as the movie distributor,
+to interrogate three categorical columns simultaneously.
+
+TODO should we include something with relative counts? aka normalize per row or column, I belive this would be a manual preprocess step or custom js in altair via transform_calculate, quite useful but maybe too much (this module is short though)
```

Comment:
Yes, please see my earlier comment that we need to teach them how to plot counts as proportions, and discuss why and when to do this. This is particularly useful with stacked bar charts.

Reply:
https://altair-viz.github.io/gallery/normalized_stacked_bar_chart.html Should maybe ideally have done this in 2 either when talking about the areas or bars there but I guess we will do it here instead. thats good isnce this is a bit shorter and doen't have to feel out of place